src/Guides/RestructuredText/Parser/DocumentParser.php

```diff
@@ -373,7 +381,11 @@ private function parseLine(string $line): bool
                     }
 
                     $this->flush();
-                    $this->initDirective($line);
+                    $this->subparser = new Subparsers\DirectiveParser($this->environment, $this->lineChecker, $this->lineDataParser, $this->directives);
+                    $directive = $this->subparser->init($line);
+                    if ($directive instanceof DirectiveHandler) {
+                        $this->directive = $directive;
+                    }
```

Comment:
Is here a return missing? when `$directive instanceof DirectiveHandler === false`? Feels like we need a reset if the line didn't match a DirectiveHandler?